consensus/state_test.go

```diff
@@ -444,6 +444,12 @@ func TestStateLockNoPOL(t *testing.T) {
 	vs2 := vss[1]
 	height, round := cs1.Height, cs1.Round
 
+	// timeouts have been increased because the CI keeps failing when the tests
+	// pass locally
+	precommitTimeout := cs1.config.Precommit(round).Nanoseconds() * 4
+	proposeTimeout := cs1.config.Propose(round).Nanoseconds() * 4
+	prevoteTimeout := cs1.config.Prevote(round).Nanoseconds() * 4
```

Comment:
@liamsi how terrible is this?

Reply:
@evan-forbes?

Reply:
It could be that the timeouts are so short, (10000000ns, 40000000ns, 10000000ns respectively) that adding any computation in a resource constrained env (github actions) that this is necessary. Tests pass locally with out the additional time, but tbh I'm not 100% sure

Reply:
Yeah, but they passed previously on CI, and now they are not, though the PR does not introduce any additional computational requirements AFAIU. That is why this does not make sense to me.

Reply:
I'm still not 100% positive that this is the cause, but there is more computation introduced each time the [blockID is validated](https://github.com/lazyledger/lazyledger-core/blob/17055a67fc101d161fde0828729b4b8972592ac3/types/block.go#L1772-L1772). Before their was no obviously no data availability header, and now a minimum data availability header has to be hashed if it hasn't already.

Reply:
and [hashing the header during validation](https://github.com/lazyledger/lazyledger-core/blob/17055a67fc101d161fde0828729b4b8972592ac3/types/block.go#L145-L147) might be a mistake.

Reply:
> and hashing the header during validation might be a mistake.

ahh ohkay, I remember now, the dah.hash field is not exported, so converting to and from proto erases it and makes a bunch of tests angry, so I added that extra hash command during validation. For the record, I really don't like this. Validating the struct should obviously not modify the struct by adding the hash. For now though, unless we want to export the hash field in `DataAvailabilityHeader`, we have to do this.

Do we want to export the hash field?

Reply:
Hmm, exporting the field would mean that we introduce something that is not spec compliant (cc @adlerjohn). the hash is something that is currently only used internally for some kind of caching. So modifying that field should happen with a call to dah.Hash() and it is only to prevent that the hash is recomputed each time. It does not necessarily need to happen during validation though.

Reply:
Instead of cs1.config.Precommit(round).Nanoseconds() * 4, you could modify the config for that test.

Reply:
> So modifying that field should happen with a call to dah.Hash() and it is only to prevent that the hash is recomputed each time. It does not necessarily need to happen during validation though.

definitely agree. The only problem arises when considering tests such as `TestMsgToProto`. It is comparing proto messages before and after encoding/decoding, during which the unexported hash field gets erased. In this case, if we want to call `dah.Hash()` we'd have to type check each message https://github.com/lazyledger/lazyledger-core/blob/17055a67fc101d161fde0828729b4b8972592ac3/consensus/msgs_test.go#L23-L221

Does this change things at all? Should we keep filling in the hash during validation?

If so, I think we can remove it come #278.

Reply:
I didn't fully look into that test but if it is just a proto concern, why not fill that field when doing the *FromProto instead?

Reply:
that seems like the best solution! https://github.com/lazyledger/lazyledger-core/pull/312/commits/10366d2b6f7f01bda263e96a82333a41f9d854ba was even able to remove the timeouts! https://github.com/lazyledger/lazyledger-core/pull/312/commits/9fd4f9f86e8a7c2f633d6f546a898eb852a95ac8

thanks, @liamsi

Reply:
Noice👌